src/nginx/proto/status.proto

```diff
@@ -106,11 +106,22 @@ message ProcessStatus {
   repeated google.api_manager.proto.EspStatus esp_status = 6;
 }
 
+// Maps service configuration IDs to their corresponding traffic percentage.
+// Key is the service configuration ID, Value is the traffic percentage
+message ServiceConfigRollouts {
+  string service_name = 1;
+  string rollout_id = 2;
+  map<string, double> percentages = 3;
+}
+
 // Top-level endpoints status message
 message Status {
   // Overall server status
   ServerStatus server = 1;
 
   // Status for each process
   repeated ProcessStatus processes = 2;
+
+  // ESP rollouts
+  repeated ServiceConfigRollouts esp_rollouts = 3;
```

Comment:
We can add the fixed length char array buffer to the share memory and store the serialized rollouts information. I think this works in most case.

<rollout_id>|<config_id count>|<config_id 1>|<percentage 1>|<config_id 2>|<percentage 2>|...|<config_id n>|<percentage n>\0

This information is updated initially and when a new rollout deployed.